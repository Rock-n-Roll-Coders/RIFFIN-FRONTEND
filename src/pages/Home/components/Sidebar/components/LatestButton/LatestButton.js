// Components / hooks
import { useNavigate, useLocation } from "react-router-dom";
import { useEffect, useState } from "react";

// MUI
import Button from '@mui/material/Button';
import StarsIcon from '@mui/icons-material/Stars';

const LatestButton = () => {
  const [variant, setVariant] = useState("text")
  const navigate = useNavigate()
  const location = useLocation()
  const handleClick = () => navigate('/latest')

  useEffect(()=> {
    if(location.pathname.startsWith("/latest")) {
      setVariant("contained")
    } else {
      setVariant("text")
    }
  }, [location])

  return (  
    <Button 
      startIcon={<StarsIcon />}
      onClick={handleClick}
      sx={{justifyContent: 'left', pl: '16px', my: 1}}
      disableElevation
      variant={variant}
    >
      Latest
    </Button>  
  );
}
 
export default LatestButton;