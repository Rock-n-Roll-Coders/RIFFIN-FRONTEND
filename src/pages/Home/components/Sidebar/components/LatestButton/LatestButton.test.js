import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import LatestButton from './LatestButton';
import {BrowserRouter, MemoryRouter} from 'react-router-dom'
import userEvent from "@testing-library/user-event";

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate
}));

test('variant is "text" when not viewing latest', ()=> {
  const component = (
    <MemoryRouter initialEntries={['/profile']}>
      <LatestButton />
    </MemoryRouter>
  )
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

test('variant changes to "contained" when viewing latest', () => {

  const component = (
    <MemoryRouter initialEntries={['/latest']}>
      <LatestButton />
    </MemoryRouter>
  )

  const tree = renderer
  .create(component)
  .toJSON();
  expect(tree).toMatchSnapshot();
})

test('routes to latest on click', async ()=> {
  const component = (
    <BrowserRouter>
      <LatestButton />
    </BrowserRouter>
  )

  render(component)
  const user = userEvent.setup()
  await user.click(screen.getByRole('button'))
  expect(mockedNavigate).toHaveBeenCalledWith('/latest');
})