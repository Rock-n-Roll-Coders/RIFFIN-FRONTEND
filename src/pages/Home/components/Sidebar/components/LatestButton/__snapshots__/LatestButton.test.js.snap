// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variant changes to "contained" when viewing latest 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-l5i4be-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="StarsIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
      />
    </svg>
  </span>
  Latest
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`variant is "text" when not viewing latest 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-13sqc32-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="StarsIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
      />
    </svg>
  </span>
  Latest
</button>
`;
